### DISCUSSION TITLE
Adobe AIR.dll - multiple files in AIR SDK - Reason?

### DISCUSSION THREAD

**Discussion by 2jfw** [airsdk/Adobe-Runtime-Support#2498] [Q&A]
I was wondering what the reason behind duplicated `Adobe AIR.dll` in the AIR SDK is?
All in all there are 4x `Adobe AIR.dll`s (two of each 32/64Bit) - but why are there duplicated copies of each DLL in folder `air-captive` ?

`runtimes\air\win\Adobe AIR\Versions\1.0`
`runtimes\air\win64\Adobe AIR\Versions\1.0`

The same files are also located under:

`\runtimes\air-captive\win\Adobe AIR\Versions\1.0`
`\runtimes\air-captive\win64\Adobe AIR\Versions\1.0`

Removing duplicates could likely save ~33MB of disk space/reduce download size..

**Comment by ajwfrost** [comment]
We've been slowly looking at this sort of thing and trying to work that out ourselves :-) The "air-captive" folder is used when packaging up a bundle so these libraries are used if you've installed an AIR application that has the captive runtime. (Although, that's just on desktop, since for Android it uses the runtime libraries from the APKs which are all under "air" rather than "air-captive" ... not very consistent!)

The idea I believe was that these captive runtimes just included the capabilities required to run the application ("end user" builds), whereas the other libraries ("SDK" builds) include other code logic that is used by some of the build tools elsewhere in the SDK. So for example, the creation of native installation packages.

This does imply that the captive files should be different/smaller than the SDK ones... I think we weren't fully aware of this when we were first trying to get everything set up and building/packaging! but we are gradually looking to get this sort of thing updated so that it's a bit more efficient again. But I've just taken a quick look on a development machine here and found that the 'end user' DLL is 3MB larger than the 'sdk' DLL, so that means I may just be wrong here! We don't use the 'end user' ones currently, but I'm curious now as to why it's bigger...

Anyway, back to your question: yes it would be possible for us currently to drop the "air-captive" folder and just copy the files in from the "air" folder, but we've left it in case we ever do start to ship the different libraries. (And in our defence, looking at the AIR 32 SDK, this also has exactly the same files in 'air' and in 'air-captive', at least on Windows..)

thanks

**Reply by 2jfw** [reply]
Alright, thanks for the input on this! I mean its not an issue anyways so its fine to keep it like it is until you've come to a point where you can safely decide what way to go
